algebra/src/main/java/org/almostrealism/collect/computations/CollectionPermute.java

```diff
@@ -38,28 +132,99 @@ public CollectionPermute(Producer<?> collection, int... order) {
 		this.order = order;
 	}
 
+	/**
+	 * Returns the memory length required for this computation.
+	 * For CollectionPermute, this is always 1 since it performs a direct index mapping
+	 * without requiring additional memory allocation.
+	 * 
+	 * @return Always returns 1, indicating minimal memory requirements for index transformation
+	 */
 	@Override
 	public int getMemLength() { return 1; }
 
+	/**
+	 * Returns the total number of elements that will be processed by this computation.
```

Comment:
This is not the total number of elements, but the total number of kernel threads that will be used.